cmd/soroban-cli/src/config/locator.rs

```diff
@@ -228,20 +245,33 @@ impl Args {
         Ok(saved_networks.chain(default_networks).collect())
     }
 
-    pub fn read_identity(&self, name: &str) -> Result<Secret, Error> {
-        Ok(KeyType::Identity
-            .read_with_global(name, &self.local_config()?)
-            .or_else(|_| name.parse())?)
+    pub fn read_identity(&self, name: &str) -> Result<Key, Error> {
+        KeyType::Identity.read_with_global(name, &self.local_config()?)
     }
 
-    pub fn key(&self, key_or_name: &str) -> Result<Secret, Error> {
-        if let Ok(signer) = key_or_name.parse::<Secret>() {
-            Ok(signer)
+    pub fn read_key(&self, key_or_name: &str) -> Result<Key, Error> {
+        if let Ok(key) = self.read_identity(key_or_name) {
+            Ok(key)
         } else {
-            self.read_identity(key_or_name)
+            Ok(key_or_name.parse()?)
+        }
+    }
+
+    pub fn get_private_key(&self, key_or_name: &str) -> Result<Secret, Error> {
```

Comment:
nit: I wonder about maybe changing this to `get_secret_key`, since it returns a `Secret` and not the private key directly? I don't feel strongly about this, just an idea!

```suggestion
    pub fn get_secret_key(&self, key_or_name: &str) -> Result<Secret, Error> {
```

Reply:
I generally prefer the term secret over private, because it's harder to mistake or mixup public vs secret, than public vs private which can look like similar words. Also abbreviations also differ conveniently, pk vs sk.

Reply:
Good points!